Library/ViewModels/CreatePasswordViewModel.swift

```diff
@@ -0,0 +1,127 @@
+import Foundation
+import Prelude
+import ReactiveSwift
+import Result
+
+public protocol CreatePasswordViewModelInputs {
+  func newPasswordTextFieldChanged(text: String?)
+  func newPasswordTextFieldDidReturn()
+  func newPasswordConfirmationTextFieldChanged(text: String?)
+  func newPasswordConfirmationTextFieldDidReturn()
+  func viewDidAppear()
+}
+
+public protocol CreatePasswordViewModelOutputs {
+  var accessibilityFocusValidationLabel: Signal<Void, NoError> { get }
+  var newPasswordTextFieldBecomeFirstResponder: Signal<Void, NoError> { get }
+  var newPasswordConfirmationTextFieldBecomeFirstResponder: Signal<Void, NoError> { get }
+  var newPasswordConfirmationTextFieldResignFirstResponder: Signal<Void, NoError> { get }
+  var validationLabelIsHidden: Signal<Bool, NoError> { get }
+  var validationLabelText: Signal<String?, NoError> { get }
+  var saveButtonIsEnabled: Signal<Bool, NoError> { get }
+
+  func currentValidationLabelText() -> String?
+}
+
+public protocol CreatePasswordViewModelType {
+  var inputs: CreatePasswordViewModelInputs { get }
+  var outputs: CreatePasswordViewModelOutputs { get }
+}
+
+public class CreatePasswordViewModel: CreatePasswordViewModelType,
+CreatePasswordViewModelInputs, CreatePasswordViewModelOutputs {
+  public init() {
+    self.newPasswordTextFieldBecomeFirstResponder = self.viewDidAppearProperty.signal
+    self.newPasswordConfirmationTextFieldBecomeFirstResponder = self.newPasswordDidReturnProperty.signal
+    self.newPasswordConfirmationTextFieldResignFirstResponder =
+      self.newPasswordConfirmationDidReturnProperty.signal
+
+    let combinedPasswords = Signal.combineLatest(
+      self.newPasswordChangedProperty.signal.skipNil(),
+      self.newPasswordConfirmationChangedProperty.signal.skipNil()
+    )
+
+    let validationMatch = combinedPasswords.map(passwordsMatch)
+    let validationLength = self.newPasswordChangedProperty.signal.skipNil().map(passwordLengthValid)
+
+    self.validationLabelText = Signal.combineLatest(validationMatch, validationLength)
```

Comment:
I think here the `combineLatest` is preventing the NewPasswordTextField to show the length validation label, since it will emit only after `validationMatch` and validationLength` emit.

Reply:
I simply copied change password behavior. I've looked into it more and have a better way of doing things...stay tuned! 😎